management/server/jwtclaims/jwtValidator.go

```diff
@@ -115,15 +128,16 @@ func NewJWTValidator(ctx context.Context, issuer string, audienceList []string,
 				}
 			}
 
-			cert, err := getPemCert(ctx, token, keys)
+			publicKey, err := getPublicKey(token, keys)
 			if err != nil {
+				log.Errorf("getPublicKey error: %s", err)
 				return nil, err
 			}
 
-			result, _ := jwt.ParseRSAPublicKeyFromPEM([]byte(cert))
-			return result, nil
+			return publicKey, nil
 		},
-		SigningMethod:       jwt.SigningMethodRS256,
+		// Canged SigningMethod to 'nil', since we can't tell yet...
+		SigningMethod:       nil, // jwt.SigningMethodRS256,
```

Comment:
We still needs to add the correct signing method here. As `nil` will invalidate this check https://github.com/netbirdio/netbird/blob/ddea0011709f091acd902d871822187f75fb091d/management/server/jwtclaims/jwtValidator.go#L165-L171

Reply:
That is true, But the point is, You can't expect that the token is signed with a specific algorithm unless the OIDC Client is registered specifically with the instruction to use that specific algorithms for token signing. If that was not explicitly instructed during client registration, then the OIDC Provider just picks one algorithm and you can see in the header the algorithm that was use. My 2-cents: The code is unneccessay strict to enforce a specific algortihm.

Reply:
we're not hardcoding the algorithm but rather extracting it dynamically from the JWK that matches the token.

For instance, we can update the `getPublicKey`  to return both the public key and the signing method:
```go
func getPublicKey(token *jwt.Token, jwks *Jwks) (interface{}, jwt.SigningMethod, error) {}
```
This allows us to validate the token using the correct signing method in `ValidationKeyGetter`, ensuring that the process is flexible and aligns with the algorithm specified in the token header.

Reply:
@bcmmbaga I am afraid I do not see your point. We could off course return the SigingMethod to look at the alg value in the jwt token. But the The ValidateKeyGetter is comparing the same 'alg' value (which is off course equal...)
Are we on the same page ?

Reply:
I understand your point now. In that case, let's go ahead and remove the `SigningMethod` as well as the check for the algo. Then we can merge this.

Reply:
Agreed, Will do !